spec/mongo/transactions_examples_spec.rb

```diff
@@ -0,0 +1,236 @@
+        hr[:employees].insert_one({ employee: 4, status: 'Active' }, session: session)
+        reporting[:events].insert_one({ employee: 4, status: { new: 'Active', old: nil } },
+                                      session: session)
+        commit_with_retry(session)
+      end
+
+      it 'makes the changes to the database' do
+        run_transaction
+        employee = hr[:employees].find({ employee: 4 }, limit: 1).first
+        expect(employee).to_not be_nil
+        expect(employee['status']).to eq('Active')
+      end
+    end
+  end
+
+  context 'Transactions Retry Example 3 (combined example)' do
+
+    let(:client) do
+      authorized_client
+    end
+
+    let(:run_transaction) do
+
+      # Start Transactions Retry Example 3
+
+      def run_transaction_with_retry(session)
+        loop do
+          begin
+            yield session # performs transaction
+            break
+          rescue Mongo::Error::OperationFailure, Mongo::Error::SocketError,
+                 Mongo::Error::NoServerAvailable => e
+
+            puts 'Transaction aborted. Caught exception during transaction.'
+            raise unless e.label?(Mongo::Session::TRANSIENT_TRANSACTION_ERROR_LABEL)
+
+            puts "#{Mongo::Session::TRANSIENT_TRANSACTION_ERROR_LABEL}, retrying transaction ..."
+            next
+          end
+        end
+      end
+
+      def commit_with_retry(session)
+        loop do
+          begin
+            session.commit_transaction
+            puts 'Transaction committed.'
+            break
+          rescue Mongo::Error::OperationFailure, Mongo::Error::SocketError,
+                 Mongo::Error::NoServerAvailable => e
+
+            if e.label?(Mongo::Session::UNKNOWN_TRANSACTION_COMMIT_LABEL)
+              puts "#{Mongo::Session::UNKNOWN_TRANSACTION_COMMIT_LABEL}, retrying commit operation ..."
+              next
+            else
+              puts 'Error during commit ...'
+              raise
+            end
+          end
+        end
+      end
+
+      # updates two collections in a transaction
+
+      def update_employee_info(session)
+        employees_coll = session.client.use(:hr)[:employees]
+        events_coll = session.client.use(:reporting)[:events]
+
+        session.start_transaction(read_concern: { level: :snapshot },
+                                  write_concern: { w: :majority })
+        employees_coll.update_one({ employee: 3 }, { '$set' => { status: 'Inactive'} },
+                                  session: session)
+        events_coll.insert_one({ employee: 3, status: { new: 'Inactive', old: 'Active' } },
+                               session: session)
+        commit_with_retry(session)
+      end
+
+      session = client.start_session
+
+      begin
+        run_transaction_with_retry(session) { |s| update_employee_info(s) }
+      rescue StandardError => e
+        # Do something with error
+        raise
+      end
+
+      # End Transactions Example 3
```

Comment:
# End Transactions Retry Example 3  instead of End Transactions Example 3

Reply:
FYI -- you can see it rendered:

https://docs-mongodbcom-staging.corp.mongodb.com/kay/foo2/core/transactions.html#retry-transaction

(ruby is under Other drop down).